modules/base/src/io/buffered/reader.fz

```diff
@@ -244,11 +244,14 @@ public read_fully (LM type : mutate) array u8 ! (reader LM) =>
 # any trailing carriage returns (ASCII 13) from the resulting strings.
 #
 public read_delimiter_full (LM type : mutate, delim String, strip_cr bool) array String ! (reader LM) =>
-  (String.from_bytes (read_fully LM))
-    .split delim
-    .map (s -> if strip_cr && s.ends_with "\r" then s.substring 0 s.byte_length-1 else s)
-    .as_array
-
+  str := (String.from_bytes (read_fully LM))
+  if str.byte_length > 0
```

Comment:
The `if` should not be necessary, I think?

Reply:
The `if` is necessary so that an empty input/file does not return an empty line.

Output of the example code from the issue without the `if`:
```
❯ printf "" | fz /tmp/read_test.fz

❯ printf "" | fz /tmp/read_test.fz 1
using read_lines: 1: 

```